modular_skyrat/modules/implants/code/augments_brain.dm

```diff
@@ -0,0 +1,59 @@
+#define LANGUAGE_IMPLANT "implant"
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor
+	name = "empathic sensor implant"
+	desc = "This won't fix your social issues, but it may help you repress them better." //not final
+	icon_state = "brain_implant_antidrop"
+	var/active = FALSE
+	var/list/stored_items = list()
+	slot = ORGAN_SLOT_BRAIN_AUG
+	var/modifies_speech = TRUE
+	
+// This should appropriately grant and remove empathy from the implant - important if you are a shadekin for some reason to not remove native languages, and allow the procs to work
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/emp_act(severity)
+	. = ..()
+	if((organ_flags & ORGAN_FAILING) || . & EMP_PROTECT_SELF)
+		return
+	
+	owner.adjust_stutter(30 SECONDS)
+	organ_flags |= ORGAN_FAILING
+	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	owner.remove_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	addtimer(CALLBACK(src, PROC_REF(reboot)), 180 / severity)
+	to_chat(owner, span_warning("You feel overwhelmed!"))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/implant_ready()
+	RegisterSignal(owner, COMSIG_MOB_SAY, PROC_REF(handle_speech))
+	owner.grant_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	to_chat(owner, span_abductor("Your mind opens to others. You can hear the thoughts of those around you, but only faintly."))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/on_mob_remove(mob/living/carbon/implant_owner)
+	. = ..()
+	organ_flags |= ORGAN_FAILING
+	UnregisterSignal(owner, COMSIG_MOB_SAY)
+	owner.remove_language(/datum/language/marish/empathy, source = LANGUAGE_IMPLANT)
+	if(QDELETED(src))
+		return
+	to_chat(implant_owner, span_abductor("Your mind closes from others. It's quiet, now."))
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/on_mob_insert(mob/living/carbon/receiver)
+	. = ..()
+	to_chat(owner, span_abductor("You begin to feel an awareness of those around you."))
+	addtimer(CALLBACK(src, PROC_REF(implant_ready)), 90)
+
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/reboot()
+	organ_flags &= ~ORGAN_FAILING
+	implant_ready()
+
+//code mostly lifted from shadekins
+	
+/obj/item/organ/internal/cyberimp/brain/empathic_sensor/proc/modify_speech(datum/source, list/speech_args)
+	if(!(organ_flags & ORGAN_FAILING))
+		INVOKE_ASYNC(source, TYPE_PROC_REF(/obj/item/organ/internal/tongue/shadekin, actually_modify_speech), source, speech_args)
```

Comment:
It's this line that's not working. Implanted people can hear just fine, but they can't speak because this proc doesn't want to do the thing.